<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>927</width>
    <height>564</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>FSM Editor &amp; Monitor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0">
    <item>
     <widget class="QSplitter" name="horizontalSplitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="handleWidth">
       <number>6</number>
      </property>
      <widget class="QTabWidget" name="projectTab">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tabProject">
        <attribute name="title">
         <string>Project</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QTreeWidget" name="projectTree">
           <column>
            <property name="text">
             <string>Element</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tabMonitor">
        <attribute name="title">
         <string>Monitor</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="labelCurrentState">
           <property name="text">
            <string>Current State: -</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTableWidget" name="tableLastValues">
           <property name="rowCount">
            <number>0</number>
           </property>
           <property name="columnCount">
            <number>2</number>
           </property>
           <column>
            <property name="text">
             <string>Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Value</string>
            </property>
           </column>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupInject">
           <property name="title">
            <string>Inject Input</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayoutInject">
            <item>
             <widget class="QLineEdit" name="lineEditInputName">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>Input name</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEditInputValue">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>Value</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="buttonInject">
              <property name="text">
               <string>Send</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="centralWidget">
       <layout class="QVBoxLayout" name="centralLayout">
        <item>
         <layout class="QHBoxLayout" name="diagramToolbarLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="addStateButton">
            <property name="text">
             <string>Add State</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addTransitionButton">
            <property name="text">
             <string>Add Transition</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addVariableButton">
            <property name="text">
             <string>Add Variable</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addInputButton">
            <property name="text">
             <string>Add Input</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addOutputButton">
            <property name="text">
             <string>Add Output</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QSplitter" name="centralSplitter">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="handleWidth">
           <number>6</number>
          </property>
          <widget class="QGraphicsView" name="graphicsViewDiagram">
           <property name="dragMode">
            <enum>QGraphicsView::RubberBandDrag</enum>
           </property>
          </widget>
            <widget class="QLabel" name="warningBar">
           <property name="minimumSize">
             <size>
               <width>0</width><height>12</height>
             </size>
           </property>
              <property name="maximumSize">
                 <size>
                <width>16777215</width><height>20</height>
                 </size>
            </property>
           <property name="styleSheet">
             <string>QLabel { background: #ffeeee; color: #aa0000; padding-left:4px; }</string>
           </property>
           <property name="alignment">
             <set>Qt::AlignVCenter|Qt::AlignLeft</set>
           </property>
           <property name="visible">
             <bool>false</bool>
           </property>
         </widget>
          <widget class="QPlainTextEdit" name="codeEditor">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>200</height>
            </size>
           </property>
           <property name="placeholderText">
            <string>State/Transition script...</string>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QScrollArea" name="scrollProperty">
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>68</width>
          <height>492</height>
         </rect>
        </property>
        <layout class="QFormLayout" name="formProperties"/>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="actionBuildRun"/>
   <addaction name="separator"/>
   <addaction name="actionConnect"/>
   <addaction name="actionDisconnect"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNew">
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save &amp;As...</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>&amp;Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="text">
    <string>&amp;Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="text">
    <string>Cu&amp;t</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>&amp;Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>&amp;Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="text">
    <string>&amp;Delete</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionBuildRun">
   <property name="text">
    <string>&amp;Build &amp;&amp; Run</string>
   </property>
   <property name="toolTip">
    <string>Compile and launch interpreter</string>
   </property>
  </action>
  <action name="actionConnect">
   <property name="text">
    <string>&amp;Connect</string>
   </property>
   <property name="toolTip">
    <string>Connect to running interpreter</string>
   </property>
  </action>
  <action name="actionDisconnect">
   <property name="text">
    <string>Dis&amp;connect</string>
   </property>
   <property name="toolTip">
    <string>Disconnect from interpreter</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
